packages/next/src/utils.ts

```diff
@@ -1896,11 +1895,9 @@ export const onPrerenderRoute =
         });
       }
 
-      // If revalidate isn't enabled we force the /404 route to be static
-      // to match next start behavior otherwise getStaticProps would be
-      // recalled for each 404 URL path since Prerender is cached based
-      // on the URL path
```

Comment:
Confirmed this comment is no longer accurate as we have fixed the cache key to be specific to the prerender path and not the visited URL path.